module/VuFind/src/VuFind/Db/Table/AuthHash.php

```diff
@@ -101,45 +101,17 @@ public function getLatestBySessionId($sessionId)
         return $this->select($callback)->current();
     }
 
-    /**
-     * Get a query representing expired sessions (this can be passed
-     * to select() or delete() for further processing).
-     *
-     * @param int $daysOld Age in days of an "expired" session.
-     *
-     * @return function
-     */
-    public function getExpiredQuery($daysOld = 2)
-    {
-        // Determine the expiration date:
-        $expireDate = time() - $daysOld * 24 * 60 * 60;
-        $callback = function ($select) use ($expireDate) {
-            $select->where->lessThan('created', date('Y-m-d H:i:s', $expireDate));
-        };
-        return $callback;
-    }
-
     /**
      * Update the select statement to find records to delete.
      *
      * @param Select $select  Select clause
      * @param int    $daysOld Age in days of an "expired" record.
-     * @param int    $idFrom  Lowest id of rows to delete.
-     * @param int    $idTo    Highest id of rows to delete.
      *
      * @return void
      */
-    protected function expirationCallback($select, $daysOld, $idFrom = null,
-        $idTo = null
-    ) {
+    protected function expirationCallback($select, $daysOld)
```

Comment:
Unless I'm missing something, all of the expirationCallback() functions are now more or less identical except for the name of the field that we're comparing the dates against. Could we move this function into the trait and use a $this->expirationDateComparisonField property to the classes (which could default to 'created', perhaps) in order to further simplify and reduce redundant code?

Reply:
That'd be nice, but we have different kinds of fields to compare. E.g. the session table has a unix timestamp, while search has a datetime. There can also be additional rules, like with search table where the row must also have saved=0.

Reply:
Ahh, of course! Too bad. We could still probably make the code less redundant with a default implementation in the trait, but I can see how that could actually make things more confusing, so I agree that it's best to leave it alone.